server/sonar-web/src/main/js/components/SourceViewer/SourceViewerBase.tsx

```diff
@@ -414,144 +417,121 @@ export default class SourceViewerBase extends React.PureComponent {
     const fromLine = lastSourceLine.line + 1;
     // request one additional line to define `hasSourcesAfter`
     const toLine = lastSourceLine.line + LINES + 1;
-    this.props
-      .loadSources(this.props.component, fromLine, toLine, this.props.branch)
-      .then(sources => {
-        this.props.loadIssues(this.props.component, fromLine, toLine).then(issues => {
-          if (this.mounted) {
-            this.setState(prevState => {
-              const nextIssues = uniqBy([...prevState.issues, ...issues], issue => issue.key);
-              return {
-                issues: nextIssues,
-                issuesByLine: issuesByLine(nextIssues),
-                issueLocationsByLine: locationsByLine(nextIssues),
-                hasSourcesAfter: sources.length > LINES,
-                loadingSourcesAfter: false,
-                sources: [
-                  ...prevState.sources,
-                  ...this.computeCoverageStatus(sources.slice(0, LINES))
-                ],
-                symbolsByLine: {
-                  ...prevState.symbolsByLine,
-                  ...symbolsByLine(sources.slice(0, LINES))
-                }
-              };
-            });
-          }
-        });
-      });
-  };
-
-  loadDuplications = (line /*: SourceLine */) => {
-    getDuplications(this.props.component, this.props.branch).then(r => {
-      if (this.mounted) {
-        this.setState(
-          {
-            displayDuplications: true,
-            duplications: r.duplications,
-            duplicationsByLine: duplicationsByLine(r.duplications),
-            duplicatedFiles: r.files
+    this.props.loadSources!(this.props.component, fromLine, toLine, this.props.branch).then(
+      sources => {
+        this.props.loadIssues!(this.props.component, fromLine, toLine, this.props.branch).then(
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Forbidden non null assertion ([typescript:S2966](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS2966))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fstas%2Fpopups&issues=AWHb6qYww6XGHKvPuQTK&open=AWHb6qYww6XGHKvPuQTK)

[ISSUE_KEY]: AWHb6qYww6XGHKvPuQTK